### PR TITLE
Switch to IRSA for AWS authentication

### PR DESCRIPTION
**Pull request by Harriethw**
# Context

**Background here:** https://dsdmoj.atlassian.net/wiki/spaces/NOM/pages/4413030459/Migrate+to+using+IRSA 
**Slack thread:** https://mojdt.slack.com/archives/C03K0HB0LBE/p1692367119011029?thread_ts=1691154904.648879&cid=C03K0HB0LBE
 
We need to switch over to using IRSA by 4th September to access AWS
services - in the case of CAS UI's this just means SQS.[1]

If I understand correctly, IRSA effectively means our kubernetes service account 
is given an IAM role which it can then use to authenticate, rather than passing in 
access keys/secrets. The service account name has been added to the cloud platform 
repo already (see below).

This PR removes the credentials we were previously using to access
SQS - from now on the kubernetes service account that this pod runs in
should have an IAM role which it will use to authenticate.

The cloud platform configuration has been merged:
dev: https://github.com/ministryofjustice/cloud-platform-environments/pull/15718/files
prod: https://github.com/ministryofjustice/cloud-platform-environments/pull/15794 
preprod: https://github.com/ministryofjustice/cloud-platform-environments/pull/15793

And CAS2 is able to use IRSA in dev now.

[1] The CAS API needs access to Domain events SNS, for which the IRSA config was already added and the backend changes made:
Prod https://github.com/ministryofjustice/cloud-platform-environments/pull/14891
PreProd https://github.com/ministryofjustice/cloud-platform-environments/pull/14890 
Dev https://github.com/ministryofjustice/cloud-platform-environments/pull/14880 

API: https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/850

# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

### CODE REVIEW

File: helm_deploy/hmpps-temporary-accommodation-ui/values.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 generic-service:
   nameOverride: hmpps-temporary-accommodation-ui
-  serviceAccountName: hmpps-community-accommodation-api-service-account
+  serviceAccountName: hmpps-temporary-accommodation-service-account
```
Comment (tahb) [inline helm_deploy/hmpps-temporary-accommodation-ui/values.yaml:4]: Looks like there's a dangling fixup! commit to autosquash.

Where does this name come from? I guess it has to match [the name we define on the platform](https://github.com/ministryofjustice/cloud-platform-environments/pull/15718/files#diff-fd4ee85b307ffdb00a1125f208a7ef5038d1a07b4d059508daa344723335e1cdR65)?

**Review by tahb — APPROVED** [review]
Thank you very much for putting this together and leading on this for us Harriet 🌟 

I think it's ready to take out of draft, autosquash the fix commit and then I can merge it and test our auditing tomorrow.

File: helm_deploy/hmpps-temporary-accommodation-ui/values.yaml (hunk 1 above)
Comment (Harriethw) [reply helm_deploy/hmpps-temporary-accommodation-ui/values.yaml:4]: fixedup!

>Where does this name come from? I guess it has to match [the name we define on the platform](https://github.com/ministryofjustice/cloud-platform-environments/pull/15718/files#diff-fd4ee85b307ffdb00a1125f208a7ef5038d1a07b4d059508daa344723335e1cdR65)?

yep exactly (there's a link to the PR in the original commit message)

**Comment by Harriethw** [comment]
@tahb thanks Tom! should be good to go now

**Pull request merged**